packages/pkl.impl.ghactions/PklCI.pkl

```diff
@@ -412,34 +421,33 @@ local withPublishTestResultsForMatrixBuild: Mixin<Workflow.Jobs> = (it) -> (it)
 local withPublishTestResults: Mixin<Workflow.Jobs> = (it) ->
   if (testReports.junit.isEmpty)
     it
-  else
+  else if (it.length > 1 || it.any((_, job) -> job.strategy?.matrix != null))
```

Comment:
Instead of `it.length`, shouldn't this count jobs where `testReports.includes(key)`?